tests/unit/python/foglamp/services/core/api/test_common_ping.py

```diff
@@ -76,40 +122,144 @@ def ssl_ctx(certs_path):
 @pytest.allure.feature("unit")
 @pytest.allure.story("api", "common")
 async def test_ping_https(test_server, ssl_ctx, test_client, loop):
+    payload = {"return": ["key", "description", "value"], "sort": {"column": "key", "direction": "asc"}}
+    result = {"rows": [
+                {"value": 1, "key": "PURGED", "description": "blah6"},
+                {"value": 2, "key": "READINGS", "description": "blah1"},
+                {"value": 3, "key": "SENT_1", "description": "blah2"},
+                {"value": 4, "key": "SENT_2", "description": "blah3"},
+                {"value": 5, "key": "SENT_3", "description": "blah4"},
+                {"value": 6, "key": "SENT_4", "description": "blah5"},
+               ]}
+
+    mockedStorageClient = MagicMock(StorageClient)
+    with patch.object(middleware._logger, 'info') as logger_info:
+        with patch.object(connect, 'get_storage', return_value=mockedStorageClient):
+            with patch.object(mockedStorageClient, 'query_tbl_with_payload', return_value=result) as query_patch:
+                app = web.Application(loop=loop, middlewares=[middleware.optional_auth_middleware])
+                # fill route table
+                routes.setup(app)
+
+                server = await test_server(app, ssl=ssl_ctx)
+                server.start_server(loop=loop)
+
+                url = 'https://127.0.0.1:{}/foglamp/ping'.format(server.port)
+                async with aiohttp.ClientSession(loop=loop) as session:
+                    async with session.get(url, verify_ssl=False) as resp:
+                        assert 200 == resp.status
+                        content = await resp.text()
+                        content_dict = json.loads(content)
+                        assert 0.0 < content_dict["uptime"]
+
+                with pytest.raises(aiohttp.ClientConnectorSSLError) as error_exec:
+                    client = await test_client(server)
+                    resp = await client.get('/foglamp/ping')
+                assert "[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed" in str(error_exec)
+
+                with pytest.raises(aiohttp.ClientConnectorSSLError) as error_exec:
+                    # self signed certificate,
+                    # and we are not using SSL context here for client as verifier
+                    connector = aiohttp.TCPConnector(verify_ssl=True, loop=loop)
+                    client = await test_client(server, connector=connector)
+                    resp = await client.get('/foglamp/ping')
```

Comment:
Please remove resp object as we are not holding any result and fix at other places as well